src/clean-core/native/memory.hh

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstddef>
+
+namespace cc
+{
+// reserves a range of pages in virtual memory
+std::byte* reserve_virtual_memory(size_t size_bytes);
```

Comment:
how would my use case of "I want 1 TB memory that is commit-on-use" work?

Reply:
On linux that's just malloc(1TB) iirc, these are for explicit commit/decommit as is done by the vmem allocs